index.bs

```diff
@@ -175,6 +175,88 @@ partial interface Navigator {
 };
 </pre>
 
+# Audio session algorithms # {#audio-session-algorithms}
+
+## Update AudioSession's type ## {#audio-session-update-type-algorithm}
+
+To <dfn>update the type</dfn> of |audioSession|, the user agent MUST run the following steps:
+1. If |audioSession|.[=AudioSession/[[isTypeBeingApplied]]=] is `true`, abort these steps.
+1. Set |audioSession|.[=AudioSession/[[isTypeBeingApplied]]=] to `true`.
+1. [=Queue a task=] to run the following steps:
+    1. Set |audioSession|.[=AudioSession/[[isTypeBeingApplied]]=] to `false`.
+    1. If |audioSession|.[=AudioSession/[[type]]=] is the same as |audioSession|.[=AudioSession/[[appliedType]]=], abort these steps.
+    1. Set |audioSession|.[=AudioSession/[[appliedType]]=] to |audioSession|.[=AudioSession/[[type]]=].
+    1. [=Update all AudioSession states=] of |audioSession|'s [=top-level browsing context=] with |audioSession|.
+    1. For each |element| of |audioSession|.[=AudioSession/[[elements]]=], [=update an element|update=] |element|.
```

Comment:
nits : `update and element` sounds vague. What does `update` mean? Do you want to refer `to run the steps of 'Update the selected audio session'? If so, it would be worth to mention the full name of the step and add a link for it.

Reply:
`update an element` is defined in the audio sink/source section which is filled in the other PR. See https://pr-preview.s3.amazonaws.com/youennf/audio-session/pull/34.html#update-an-element

The principle is that, when the audio session changes, the elements of the audio session might need to be updated.